flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/DatasetSink.java

```diff
@@ -219,52 +249,47 @@ public synchronized void stop() {
   @Override
   public Status process() throws EventDeliveryException {
     if (writer == null) {
-      try {
-        this.writer = newWriter(login, target);
-      } catch (DatasetException e) {
-        // DatasetException includes DatasetNotFoundException
-        throw new EventDeliveryException(
-          "Cannot write to " + getName(), e);
-      }
+      this.writer = newWriter();
     }
 
-    // handle file rolling
-    if ((System.currentTimeMillis() - lastRolledMs) / 1000 > rollIntervalS) {
-      // close the current writer and get a new one
-      writer.close();
-      this.writer = newWriter(login, target);
-      this.lastRolledMs = System.currentTimeMillis();
-      LOG.info("Rolled writer for " + getName());
+    if (shouldRoll()) {
+      rollWriter(true, true);
     }
 
     Channel channel = getChannel();
-    Transaction transaction = null;
     try {
       long processedEvents = 0;
 
-      transaction = channel.getTransaction();
-      transaction.begin();
+      // TODO: Can the same Sink object be called from multiple threads
```

Comment:
Ok, so that means that `stop()` can be called while `process()` is still happening so I need a way to make sure that `process()` finishes before committing the transaction in stop(). Correct?